src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1567 @@
+            of `highdicom.base.SOPClass`
+
+        """  # noqa: E501
+        if sop_class_uid not in (
+            GrayscaleSoftcopyPresentationStateStorage,
+            PseudoColorSoftcopyPresentationStateStorage,
+            ColorSoftcopyPresentationStateStorage
+        ):
+            raise ValueError(
+                'The IOD indicated by the provided sop_class_uid is not '
+                'implemented.'
+            )
+
+        # Check referenced images are from the same series and have the same
+        # size
+        ref_series_uid = referenced_images[0].SeriesInstanceUID
+        ref_im_rows = referenced_images[0].Rows
+        ref_im_columns = referenced_images[0].Columns
+        for ref_im in referenced_images:
+            series_uid = ref_im.SeriesInstanceUID
+            if series_uid != ref_series_uid:
+                raise ValueError(
+                    'Images belonging to different series are not supported.'
+                )
+            if ref_im.Rows != ref_im_rows or ref_im.Columns != ref_im_columns:
+                raise ValueError(
+                    'Images with different sizes are not supported.'
+                )
+
+            if sop_class_uid == ColorSoftcopyPresentationStateStorage:
+                if ref_im.SamplesPerPixel != 3:
+                    raise ValueError(
+                        'For color presentation states, all referenced '
+                        'images must have 3 samples per pixel.'
+                    )
+            else:
+                if ref_im.SamplesPerPixel != 1:
+                    raise ValueError(
+                        'For grayscale and pseduo-color presentation states, '
+                        'all referenced images must have a single sample '
+                        'per pixel.'
+                    )
+
+        super().__init__(
+            study_instance_uid=referenced_images[0].StudyInstanceUID,
+            series_instance_uid=series_instance_uid,
+            series_number=series_number,
+            sop_instance_uid=sop_instance_uid,
+            sop_class_uid=sop_class_uid,
+            instance_number=instance_number,
+            manufacturer=manufacturer,
+            modality='PR',
+            patient_id=referenced_images[0].PatientID,
+            transfer_syntax_uid=transfer_syntax_uid,
+            patient_name=referenced_images[0].PatientName,
+            patient_birth_date=referenced_images[0].PatientBirthDate,
+            patient_sex=referenced_images[0].PatientSex,
+            accession_number=referenced_images[0].AccessionNumber,
+            study_id=referenced_images[0].StudyID,
+            study_date=referenced_images[0].StudyDate,
+            study_time=referenced_images[0].StudyTime,
+            referring_physician_name=getattr(
+                referenced_images[0], 'ReferringPhysicianName', None
+            ),
+            **kwargs
+        )
+        self.copy_patient_and_study_information(referenced_images[0])
```

Comment:
Makes sense to me, but what do you want to do in the short term given that the module is not there in the standard? I'm hesitant to put that information in just yet if it's not compliant

Reply:
Unfortunately, the Specimen IE is still not yet a first-class citizen and the Specimen module didn't  make it into corners of the standard where pathology folks have not yet wandered. The same was the case for Segmentation and Parametric Map. I should be included in my opinion. cc @dclunie

Please already include it. Even if the module is not specified in the standard, adding additional attributes won't necessarily violate the standard either as far as I understand it. It would just be considered an extension of the standard for the time being.

Reply:
Addressed in 474999f